R/gen_msf_data.R

```diff
@@ -857,6 +857,123 @@ gen_msf_data <- function(dictionary, dat_dict, is_survey, varnames = "data_eleme
 
   }
 
+  if (dictionary == "ebs") {
+
+    # set date of event starting to the earliest date from those given
+    dis_output$date_event_start <- with(
+      dis_output,
+      pmin(
+        date_event_start,
+        date_signal,
+        date_triage,
+        date_verification,
+        date_assessment,
+        date_response_started,
+        date_response_ended,
+        na.rm = TRUE
+      )
+    )
+
+    # date signal
+    dis_output <- enforce_timing(dis_output,
+                                 first  = "date_signal",
+                                 second = "date_triage",
+                                 5:30
+    )
+
+    ## date triage
+    dis_output <- enforce_timing(dis_output,
+                                 first  = "date_triage",
+                                 second = "date_verification",
+                                 5:30,
+                                 inclusive = TRUE
+    )
+
+    # date verification
+    dis_output <- enforce_timing(dis_output,
+                                 first  = "date_verification",
+                                 second = "date_assessment",
+                                 5:30
+    )
+
+    ## date assessment
+    dis_output <- enforce_timing(dis_output,
+                                 first  = "date_assessment",
+                                 second = "date_response_started",
+                                 5:30,
+                                 inclusive = TRUE
+    )
+
+    ## date response started
+    dis_output <- enforce_timing(dis_output,
+                                 first  = "date_response_started",
+                                 second = "date_response_ended",
+                                 5:30,
+                                 inclusive = TRUE
+    )
+
+    ## create an initials variable of 3 random letters
+    dis_output$initials <- stringi::stri_rand_strings(nrow(dis_output), 2,
```

Comment:
instead of stringi please use base as below - this way less dependencies and less problems for me to fix going forward 
`
replicate(nrow(dis_output), paste0(sample(letters, 2, replace = TRUE), collapse = ""))`